pydatastructs/trees/heaps.py

```diff
@@ -0,0 +1,223 @@
+from pydatastructs.utils.misc_util import _check_type
+
+NoneType = type(None)
+
+__all__ = [
+        'BinaryHeap'
+]
+
+class BinaryHeap():
+    """
+    Respresentation of BinaryHeap data structure
+
+    Parameters
+    ==========
+
+    array : list
+        Optional, by default 'None'
+        List of initial elements in Heap
+        
+    _type : str

```

Comment:
Why _?

Reply:
I thought that as type is manually given by the user so the user knows the type of heap. So made it private.